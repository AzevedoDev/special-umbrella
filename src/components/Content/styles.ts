import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.main`
  ${({ theme }) => css`
    background-color: ${theme.colors.mainBg};
    height: 100%;
    width: 100%;
    font-family: ${theme.font.family};
    z-index: ${theme.layers.menu};
    ${media.lessThan('medium')`
      background-color: ${theme.colors.mainBg};
      margin-top: -6.3rem;
      clip-path: ellipse(180% 100% at bottom);
    `}
    ${media.greaterThan('medium')`
      background-color: ${theme.colors.mainBg};
      margin-left: -9.3rem;
      clip-path: ellipse(100% 150% at right);
      display:flex;
      flex-direction: column;
      justify-content:flex-end;
      align-items: center;
      padding-bottom: 9.7rem;
    `}
    ${media.greaterThan('large')`
      background-color: ${theme.colors.mainBg};
      margin-left: -9.3rem;
      clip-path: ellipse(100% 160% at right);
      display:flex;
      flex-direction: column;
      justify-content:flex-end;
      align-items: center;
      padding-bottom: 9.7rem;
    `}
  `}
`
export const DrawnLotteryNumbersWrapper = styled.div`
  display: flex;
  width: 100%;
  height: 100%;
  padding-top: 10rem;
  ${media.greaterThan('medium')`
    flex-direction: row;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
    max-height: 20rem;
    padding: 0 2.3rem 0 5rem;
  `}
  ${media.lessThan('medium')`
    padding-bottom: 8.4rem;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
  `}
`

export const TextWrapper = styled.div`
  ${media.greaterThan('medium')`
    width: 100%;
    display: flex;
    padding: 0 2.3rem 0 5rem;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    text-align: center;
  `}

  ${media.lessThan('medium')`
    width: 31.5rem;
    display: flex;
    flex-direction: column;
    padding: 0 2rem;
  `}
`
export const MobileGroup = styled.div`
  display: none;
  ${({ theme }) => css`
    ${media.lessThan('medium')`
      background-color: ${theme.colors.mainBg};
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      align-items: center;
      padding: 3.8rem;
      width: 100%;
    `}
  `}
`
export const DesktopGroup = styled.div`
  display: none;
  ${({ theme }) => css`
    ${media.greaterThan('medium')`
      background-color: ${theme.colors.mainBg};
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      align-items: center;
      width: 100%;
      height: 60%;
    `}
  `}
`
