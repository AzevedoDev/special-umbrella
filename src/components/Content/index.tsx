import DrawnLotteryNumber from 'components/DrawnLotteryNumber'
import Text from 'components/Text'
import { Concurso } from 'graphql/generated/graphql'
import * as S from './styles'

export type ContentProps = {
  concurso: Concurso
}

const Content = ({ concurso }: ContentProps) => (
  <S.Wrapper>
    <S.MobileGroup>
      <S.DrawnLotteryNumbersWrapper>
        {concurso.numeros?.map((lotteryNumber: string | null | undefined) => (
          <DrawnLotteryNumber
            key={lotteryNumber}
            lotteryNumber={lotteryNumber}
          />
        ))}
      </S.DrawnLotteryNumbersWrapper>
      <S.TextWrapper>
        <Text color="black">
          Este sorteio é meramente ilustrativo e não possui nenhuma ligação com
          a CAIXA.
        </Text>
      </S.TextWrapper>
    </S.MobileGroup>
    <S.DesktopGroup>
      <S.DrawnLotteryNumbersWrapper>
        {concurso.numeros?.map((lotteryNumber: string | null | undefined) => (
          <DrawnLotteryNumber
            key={lotteryNumber}
            lotteryNumber={lotteryNumber}
          />
        ))}
      </S.DrawnLotteryNumbersWrapper>
      <S.TextWrapper>
        <Text color="black">
          Este sorteio é meramente ilustrativo e não possui nenhuma ligação com
          a CAIXA.
        </Text>
      </S.TextWrapper>
    </S.DesktopGroup>
  </S.Wrapper>
)

export default Content
