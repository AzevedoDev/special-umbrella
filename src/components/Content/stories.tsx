import { Story, Meta } from '@storybook/react/types-6-0'
import Content, { ContentProps } from '.'
import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 6.3rem;
`

export default {
  title: 'Content',
  component: Content,
  args: {
    concurso: {
      id: '1',
      numeros: ['01', '01', '01', '01', '01', '01']
    }
  }
} as Meta

export const Default: Story<ContentProps> = (args) => (
  <Wrapper>
    <Content {...args} />
  </Wrapper>
)
