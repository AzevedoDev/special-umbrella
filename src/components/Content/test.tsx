import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Content from '.'
const concurso = {
  id: '5534',
  loteria: 1,
  numeros: ['17', '32', '41', '58', '69'],
  data: '2021-04-26T21:12:23.562Z'
}

describe('<Content />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Content concurso={concurso} />)

    expect(
      screen.getByText(
        /Este sorteio é meramente ilustrativo e não possui nenhuma ligação com a CAIXA/i
      )
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
