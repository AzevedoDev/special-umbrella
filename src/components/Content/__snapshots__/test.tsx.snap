// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> should render the heading 1`] = `
.c2 {
  background-color: #FFFFFF;
  width: 7.6rem;
  height: 7.6rem;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1rem;
  font-family: 'Montserrat',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
}

.c3 {
  color: #333333;
  font-size: 2rem;
  font-weight: 700;
}

.c5 {
  font-size: 1.4rem;
  color: #333333;
  font-weight: 500;
  text-align: center;
  font-family: 'Montserrat',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
}

.c0 {
  background-color: #EFEFEF;
  padding-top: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-clip-path: ellipse(180% 100% at bottom);
  clip-path: ellipse(180% 100% at bottom);
  margin-top: -6.3rem;
  padding: 3.8rem;
  font-family: 'Montserrat',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  height: 100%;
  padding-bottom: 8.4rem;
}

.c4 {
  width: 31.5rem;
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <main
      class="c2"
    >
      <p
        class="c3"
      >
        17
      </p>
    </main>
    <main
      class="c2"
    >
      <p
        class="c3"
      >
        32
      </p>
    </main>
    <main
      class="c2"
    >
      <p
        class="c3"
      >
        41
      </p>
    </main>
    <main
      class="c2"
    >
      <p
        class="c3"
      >
        58
      </p>
    </main>
    <main
      class="c2"
    >
      <p
        class="c3"
      >
        69
      </p>
    </main>
  </div>
  <div
    class="c4"
  >
    <p
      class="c5"
      color="black"
    >
      Este sorteio é meramente ilustrativo e não possui nenhuma ligação com a CAIXA.
    </p>
  </div>
</main>
`
